packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -854,6 +855,12 @@ class SemanticsProperties extends DiagnosticableTree {
   /// This is mutually exclusive with [toggled].
   final bool? checked;
 
+  /// If true, indicates that this subtree represents a checkbox
+  /// or similar widget with a "mixed" check state.
+  ///
+  /// This must be false when [Checkbox.tristate] is false.
```

Comment:
Just curious, when can this be `null`? If never, should this be non-nullable?

Reply:
My thought process was to leave it to default to `null` when the widget has no checked state or is not tristate, like how `checked` is nullable and is not set when the widget has no check state.

Reply:
I'd tweak the comment to clarify the non-null case as that's not readily apparent. Effectively: non-null if this is a checkbox, true if this is a tristate checkbox in the mixed state, false otherwise.

Perhaps something like:

>  If non-null, indicates that this subtree represents a checkbox or similar widget with a "checked" state. If true, this is a tristate checkbox in the mixed state.
>
> This must be false when [Checkbox.tristate] is false.